app/workers/deletes.php

```diff
@@ -142,6 +222,49 @@ protected function deleteById(Document $document, Database $database, callable $
         Authorization::reset();
     }
 
+    protected function deleteForProjectIds(callable $callback)
+    {
+        $count = 0;
+        $chunk = 0;
+        $limit = 50;
+        $projects = [];
+        $sum = $limit;
+
+        $executionStart = \microtime(true);
+        
+        while($sum === $limit) {
+            $chunk++;
+
+            Authorization::disable();
+            $projects = $this->getConsoleDB()->getCollection([
+                'limit' => $limit,
+                'offset' => 0,
```

Comment:
If the offset is not changing you don't move forward in the list, and will end in an endless loop. Unlike the original code, here projects don't get deleted and the list isn't gettin smaller on every iteration.

Reply:
Changing the limit with no offset will only return a bigger list at every loop. Old items + new items.

Reply:
Yes I thought about it too. I referred to the `deleteByGroup()` function which had this same issue. I will fix it in both places now.